### PR TITLE
Feature/221259 rebranding changes

### PR DESCRIPTION
**Pull request by sppanday**
This pull request brings changes for 221259 and 221285

### CODE REVIEW

File: feature-communicate-list.md [hunk 1]
```diff
@@ -317,3 +317,14 @@ ACT government logo
 A secondary image has been used and read from config value secondaryLogoUrl. This uses standard ACT Govt Logo which is horizontal.
 ACT government logo on pre login screen
 Vertical ACT government logo has been used for this purpose. A config value cant be used for this as app-web is not query that setting in front end. Only on react sdk you can use config to retrieve that value. This one uses ACT gov logo that ends with _vertical keyword.
+
+Feature: 221259
+Web: Sign In Screen Lingo Re-Branding and Update
+====================================================================================
+Technical Implementation:
+Note: These configurable parameters are stored in config-communicate.json and replaced in config.json file when app loads.
+1) Homeserver edit is controlled by configuration item disable_custom_urls which when used to true can lead to homeserver be uneditable which means read only text is displayed.
```
Comment (jrschremmer) [inline feature-communicate-list.md]: Please use positively named config parameters e.g. enable_custom_homeserver_urls.
Looking at the implementation I can see that you ended up doing showHomeServerDetail, which is good, but please update this file to match the actual implementation.

File: feature-communicate-list.md [hunk 2]
```diff
@@ -317,3 +317,14 @@ ACT government logo
...
+2) Splash screen menu items UIFeature.PasswordReset and UIFeature.Registration were part of default matrix config values which has been set to true from our side to disable password reset and client's ability to create account and it will hide these two menu from splash screen.
+3) In order to hide phone number drop down menu, homeserver detail display, i-information changedSplashScreenMenuItem has been introduced in config which when value is populated its meant to control what needs to be hidden when used appropriately in implementation.
+4) Matrix default login screen before form based login screen where user would see sign in button and then sign in form would be triggered was controlled by welcome_page routing configuration in codebase which has been turned off and instead we redirect user from welcome screen in authenticated page to form based login screen with configurable parameter auto_redirect_from_welcome_screen_to_login.
```
Comment (jrschremmer) [inline feature-communicate-list.md]: I'd prefer this to be controlled by if "Create Account" e.g. UIFeature.Registration is enabled.  If it isn't then the only functionality left on this page is Sign In and Language selection, which are both on the Login/Sign In page, in which case the welcome screen can be skipped

File: src/components/structures/HomePage.tsx [hunk 3]
```diff
@@ -109,16 +109,18 @@ const HomePage: React.FC<IProps> = ({ justRegistered = false }) => {
...
         let logoUrlSecondary = SettingsStore.getValue(UIFeature.SecondaryLogoUrl);
+        let logoSecondaryDescription = config['secondary_logo_description'];
         if (brandingConfig && brandingConfig.authHeaderLogoUrl) {
             logoUrl = brandingConfig.authHeaderLogoUrl;
         }
-       const logoUrlSecondaryStyle = {
-           margin: '0 10px'
+       const logoSecondaryStyle = {
+           margin: '0 10px',
+           height: '55px'
```
Comment (jrschremmer) [inline src/components/structures/HomePage.tsx]: I'd suggest that 55px should be configuration driven?

File: src/components/views/elements/ServerPicker.tsx [hunk 4]
```diff
@@ -81,7 +82,9 @@ const ServerPicker = ({ title, dialogTitle, serverConfig, onServerConfigChange }
         </span>;
     }
 
-    return <div className="mx_ServerPicker">
+
+    return config.showHomeServerDetail &&
```
Comment (jrschremmer) [inline src/components/views/elements/ServerPicker.tsx:86]: This need to change to handle that showHomeServerDetail should default to true if showHomeServerDetail is not in the config

File: src/utils/WellKnownUtils.ts [hunk 5]
```diff
@@ -63,7 +63,7 @@ export function isE2EEEnabledInWellKnown(): boolean {
 }
 
 export function isE2EEDefaultedOnInWellKnown(): boolean {
-	// If encryption is disabled then don't default encrpyion on
+	// If encryption is disabled then don't default encryption
```
Comment (jrschremmer) [inline src/utils/WellKnownUtils.ts]: I'm not sure why this change is required, as the comment now doesn't make sense.

File: src/config.ts [hunk 6]
```diff
@@ -0,0 +1,31 @@
+import SdkConfig from "./SdkConfig";
+
+const config = SdkConfig.get();
+
+const hiddenSplashScreenMenuItem = config['hiddenSplashScreenMenuItem'];
```
Comment (jrschremmer) [inline src/config.ts]: hiddenSplashScreenMenuItem is not mentioned in feature-communicate-list.md.  I'd prefer this to be positively named and as these items aren't in a menu, perhaps call it showLoginScreenItems

File: src/config.ts [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+    if (signInTextNeedsToBeReplaced) {
+        document.querySelector('label').innerHTML = document.querySelector('label').innerHTML.replace(/Sign in/g, "Log in");
+        document.querySelector('input').innerHTML = document.querySelector('input').innerHTML.replace(/Sign in/g, "Log in");
+        document.querySelector('h2').innerHTML = document.querySelector('h2').innerHTML.replace(/Sign in/g, "Log in");
+     //   document.querySelector('form').innerHTML = document.querySelector('.mx_SSOButtons').innerHTML.replace(/Sign in/g, "Log in");
+    }
+}
+
+//Phone based authentication drop down menu
+export const allowPhoneBasedAuthentication = hiddenSplashScreenMenuItem?.find(x => x.includes(['allowPhoneBasedAuthentication'])) ? false: true;
```
Comment (jrschremmer) [inline src/config.ts]: Does this return true if hiddenSplashScreenMenuItem doesn't exist?

File: feature-communicate-list.md (hunk 1 above)
Comment (sppanday) [reply feature-communicate-list.md]: This would be out of box flag so that flag value set to "true" in config would turn homeserver text to turn into read only mode ( uneditable ) . If it was our own config then we would prefer positive I agree with you.

Comment (jrschremmer) [reply feature-communicate-list.md]: In that case, can you please change the text to say 
"Homeserver edit is controlled by existing configuration item"

File: src/config.ts (hunk 7 above)
Comment (sppanday) [reply src/config.ts]: Yeah, that was biggest part of my experiment so if you would remove hiddenSplashScreenMenuItem array from config 
 entirely then screen would return to matrix-default which is how I had checked reverse condition so it passed that test.

File: src/utils/WellKnownUtils.ts (hunk 5 above)
Comment (sppanday) [reply src/utils/WellKnownUtils.ts]: Change has been reverted. I was confused little bit initially.

File: src/components/structures/HomePage.tsx (hunk 3 above)
Comment (sppanday) [reply src/components/structures/HomePage.tsx]: May I know for my curiosity why would we want only height to be controlled from config not other styles  ? Second question would be since secondary logo is not in matrix this value wont apply if secondary logo is not present on UI so I am bit confused about height needing to be configurable ? Please enlighten me .

File: feature-communicate-list.md (hunk 2 above)
Comment (sppanday) [reply feature-communicate-list.md]: What I had found while debugging was matrix intended to create welcome screen to allow client to provide a welcoming screen  with information on UI such as "what this app does and etc with a button to take user to login screen after that" and html page is configurable for what we want to display on that screen. if we have html page in future that we want to show in front screen before login page so I was thinking welcome page is independent of create account/Registration etc. We could use Create Account flag but for me that one controlling welcome page did not look right. Please enlighten me if you disagree. Welcome page is for info of app for future audience according to matrix and login page is actual login page. I would prefer to not user registration condition be used for this if you would not mind.

File: src/config.ts (hunk 6 above)
Comment (sppanday) [reply src/config.ts]: As without using this array would show everything anyways so I thought if config is used then it is to hide so I used array called "hidden". Either way, not much difference so I am happy to change this if you have strong inclination on positive. If "shown.." is used then that would tell developer "oh only values in that array from config are shown and rest are hidden", but if "hidden .." is used then that would say "oh those items in config are hidden so I tried to make sense.  Joseph had used one negative array flag as well previously if you see in config but either way I dont mind let me know if you still prefer positive.

File: feature-communicate-list.md (hunk 2 above)
Comment (sppanday) [reply feature-communicate-list.md]: There is also possibility in matrix to replace authenticated page if we got one after login so these two pages are customisable. For me its is making better sense to say welcome page than relying on login button on login page. Let me know what you think.

Comment (jrschremmer) [reply feature-communicate-list.md]: Considering the possible additional functionality on the welcome screen, I'm happy to keep the separate config to skip the welcome screen.

File: src/components/structures/HomePage.tsx (hunk 3 above)
Comment (jrschremmer) [reply src/components/structures/HomePage.tsx]: I'm happy for all styles to come from the config ... I was just singling out something that would easily change, for other users with a different sized secondary logo.

File: src/config.ts (hunk 6 above)
Comment (jrschremmer) [reply src/config.ts]: I can live with hideLoginScreenItems

File: src/config.ts (hunk 7 above)
Comment (jrschremmer) [reply src/config.ts]: Ok, good to know.  Can you please add what ?. does in a code comment.  As I have done some google searches and haven't quickly found what this syntax does.  In the design document, can you please list this type of testing that you have performed.

Comment (sppanday) [reply src/config.ts]: Explanation has been added now in  Kanban, feature doc, and on config.ts.  Also hiddenSplash....has been renamed to your suggested one.

File: src/config.ts (hunk 6 above)
Comment (sppanday) [reply src/config.ts]: As discussed on teams chat, loginScreen has been used as updated config.

File: src/components/structures/HomePage.tsx (hunk 3 above)
Comment (sppanday) [reply src/components/structures/HomePage.tsx]: This suggestion has been implemented. This took me 2hrs now and I have implemented in a way that will make everything related to secondaryLogo configurable such as img location, margin, height, description of img in single spot. UIFeature.secondaryLogoUrl has been removed as new config is going to centralise everything now on for secondary image. I am happier than before now. Thanks for reminder as with this change I have centralised all values needed into one config value.

**Review by jrschremmer — APPROVED** [review]
